### PR TITLE
python3Packages.torchtnt: init at 0.2.4 

### PR DESCRIPTION
**Pull request by nim65s**
This add https://github.com/pytorch/tnt, a "lightweight library for PyTorch training tools and utilities".

While here, this disable a recently failing test breaking torchsnapshot, ref. https://hydra.nixos.org/build/278371025/log/tail cc @GaetanLepage 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by nim65s** [comment]
thanks @GaetanLepage for the help !

**Comment by nim65s** [comment]
let's draft this until https://github.com/NixOS/nixpkgs/pull/361813

**Comment by nim65s** [comment]
Hum, something broke in unit tests. I'll have a look later.

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 361470`

---
### `x86_64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.torchtnt</li>
    <li>python311Packages.torchtnt.dist</li>
    <li>python312Packages.torchtnt</li>
    <li>python312Packages.torchtnt.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.torchtnt</li>
    <li>python311Packages.torchtnt.dist</li>
    <li>python312Packages.torchtnt</li>
    <li>python312Packages.torchtnt.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`

---
### `aarch64-darwin`

**Comment by GaetanLepage** [comment]
> Hum, something broke in unit tests. I'll have a look later.

```
Disabled tests path "examples/torchrec/tests/torchrec_example_test.py" does not exist. Aborting
```

**Comment by nim65s** [comment]
> Hum, something broke in unit tests. I'll have a look later.

Ok, that was because tests are not included in PyPI archive

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 361470`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.torchtnt</li>
    <li>python311Packages.torchtnt.dist</li>
    <li>python312Packages.torchtnt</li>
    <li>python312Packages.torchtnt.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.torchtnt</li>
    <li>python311Packages.torchtnt.dist</li>
    <li>python312Packages.torchtnt</li>
    <li>python312Packages.torchtnt.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.torchtnt</li>
    <li>python311Packages.torchtnt.dist</li>
    <li>python312Packages.torchtnt</li>
    <li>python312Packages.torchtnt.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.torchtnt</li>
    <li>python311Packages.torchtnt.dist</li>
    <li>python312Packages.torchtnt</li>
    <li>python312Packages.torchtnt.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/torchtnt/default.nix [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  pythonImportsCheck = [
+    "torchtnt"
+  ];
+
+  # Tests are not included in Pypi archive
+  doCheck = false;
+
+  meta = {
+    description = "lightweight library for PyTorch training tools and utilities";
```
Comment (natsukium) [inline pkgs/development/python-modules/torchtnt/default.nix:66]:
```suggestion
    description = "Lightweight library for PyTorch training tools and utilities";
```
nit pick

**Review by natsukium — APPROVED** [review]
LGTM

**Pull request merged**